lib/components/hikingmapbox.dart

```diff
@@ -50,12 +65,21 @@ class _MapWidgetState extends State<MapWidget> {
   @override
   initState() {
     super.initState();
-    _loadOfflineTiles();
+
+    _loadOfflineTiles().then((result) => {
+          requestLocationPermissionIfNotAlreadyGranted().then((result) {
+            getCurrentLocation().then((location) {
+              initRoutes();
+            });
+            updateCurrentLocationOnChange();
```

Comment:
when the map is created you can receive a callback where you start tasks like:
```
getCurrentLocation()
initRoutes()
updateCurrentLocationOnChange()
```

This callback is already implemented see line 445 in the file onMapCreated.